src/core/vdom.ts

```diff
@@ -698,8 +699,11 @@ function addNodeToMap(id: string, key: string | number, node: HTMLElement) {
 	}
 }
 
-function destroyHandles(destroyMap: Map<string, () => void>) {
-	destroyMap.forEach((destroy) => destroy());
+function destroyHandles(destroyMap: Map<string, () => void>, middlewareIds: string[]) {
```

Comment:
(optional) since adding the `middlewareIds` this method has become less generic. We could refactor it as `destroyHandles(meta: WidgetMeta)` and move the `destroyMap` existence check internal.

Reply:
@devpaul updated :+1: thanks